### PR TITLE
Output recursive

### PR DESCRIPTION
**Pull request by 1stBitcoinSent**
Add `/r/output/ID` recursive endpoint.

😁😅 many 1sts for me here so please go easy on me.

### CODE REVIEW

**Comment by 1stBitcoinSent** [comment]
😀 Hi @casey Hello @raphjaph I'm not sure how to run these checks 😅

**Comment by 1stBitcoinSent** [comment]
test example using output sat_ranges
1stBitcoinSent test example

thinking 🤔 it would also be interesting to be able to represent the history of sats artistically with recursive endpoint access to transactions

**Comment by raphjaph** [comment]
How would inscriptions get the outpoint (other than hardcoding it)?

**Comment by 1stBitcoinSent** [comment]
> How would inscriptions get the outpoint (other than hardcoding it)?

Hello 😁 @raphjaph 
This is my current workflow:
`const getInscriptionId = () => window.location.pathname.split("/").pop();`
`let response = await fetch(`/r/inscription/${getInscriptionId()}`);`
`let inscriptionData = await response.json();`
`inscriptionData.output`

**Comment by raphjaph** [comment]
This definitely needs some tests though as well.

**Comment by 1stBitcoinSent** [comment]
> This definitely needs some tests though as well.

Hey @raphjaph 😁 Thank you for getting back to me on this 😀 is this ok?

**Comment by raphjaph** [comment]
I'm looking at this today, just added some structs and gonna figure out what exactly to return

**Comment by 1stBitcoinSent** [comment]
> I'm looking at this today, just added some structs and gonna figure out what exactly to return

Thank you so much raph 🧡😆

File: docs/src/inscriptions/recursion.md [hunk 1]
```diff
@@ -3202,6 +3202,37 @@ curl -s -H "Accept: application/json" \
...
+Information about an output.
+
+### Example
+```bash
+curl -s -H "Accept: application/json" \
+  http://0.0.0.0:80/r/output/4a5e1e4baab89f3a32518a88c31bc87f618f76673e2cc77ab2127b7afdeda33b:0
+```
+```json
+{
+  "address": "04678afdb0fe5548271967f1a67130b7105cd6a828e03909a67962e0ea1f61deb649f6bc3f4cef38c4f35504e51ec112de5c384df7ba0b8d578a4c702b6bf11d5f",
```
Comment (raphjaph) [inline docs/src/inscriptions/recursion.md:3222]: This is not an address.

File: docs/src/inscriptions/recursion.md [hunk 2]
```diff
@@ -3202,6 +3202,37 @@ curl -s -H "Accept: application/json" \
...
+  "indexed": true,
+  "value": 5000000000,
+  "inscriptions": [],
+  "runes": {},
```
Comment (raphjaph) [inline docs/src/inscriptions/recursion.md:3226]: we should find an example that includes some runes

File: docs/src/inscriptions/recursion.md (hunk 1 above)
Comment (1stBitcoinSent) [reply docs/src/inscriptions/recursion.md:3222]: yea it's strange 🤔
[https://mempool.space/tx/4a5e1e4baab89f3a32518a88c31bc87f618f76673e2cc77ab2127b7afdeda33b#flow=&vout=0](https://mempool.space/tx/4a5e1e4baab89f3a32518a88c31bc87f618f76673e2cc77ab2127b7afdeda33b#flow=&vout=0)

File: docs/src/inscriptions/recursion.md (hunk 2 above)
Comment (1stBitcoinSent) [reply docs/src/inscriptions/recursion.md:3226]: do you know of an output that should never change like the genesis output?
maybe a burnt rune output? would that still show a rune amount if it was burnt?

**Comment by raphjaph** [comment]
Hey, thanks for this PR but we decided to go with a slightly different approach in #4148. It was just too different to continue work here so I made a new PR.

**Pull request closed without merging**